scripts/bulk-password-update-export.js

```diff
@@ -95,40 +116,42 @@ const generatePassword = async () => {
 };
 
 // thanks https://bobbyhadz.com/blog/javascript-read-file-into-array !
-const loadUsers = async () => {
+const loadUsers = async (use_passes) => {
   try {
-    const contents = await fsPromises.readFile('user-password-change.txt', 'utf-8');
-    const lines = contents.split(/\r?\n/);
-    return lines.map(line => line.trim()).filter(line => line.length > 0);
+    const contents = await fsPromises.readFile('user-password-change.csv', 'utf-8');
+    let columns = ['name'];
+    if ( use_passes ){
+      columns = ['name', 'pass'];
+    }
+    return csvSync.parse(contents, {
+      columns: columns,
+      trim: true,
+      skip_empty_lines: true
+    });
   } catch (err) {
-    console.log(err);
+    console.log(err.message);
     process.exit(1);
   }
 };
 
-const changeUserPass = async (user, postOptions) => {
-  const newPass = await generatePassword();
-  postOptions.body = {
-    password: newPass
-  };
-  postOptions.uri = `${options.uri}/${user}`;
-  try {
-    await rpn.post(postOptions);
-    console.log('SUCCESS', user, newPass);
-  } catch (e) {
-    console.log('ERROR', user, e.message);
-  }
-};
-
 const execute = async () => {
-  const users = await loadUsers();
-  const input = await areYouSure(users.length);
+  let use_passes = false;
+  let extraWarning = '\nThey will each get a random password which will be printed below.\n';
+  if (argv.use_passes === 'true') {
+    use_passes = true;
+    extraWarning = '\nThey will have their password updated to the one in "user-password-change.csv".\n';
+  }
+  const users = await loadUsers(use_passes);
+  const input = await areYouSure(users.length, extraWarning);
   if (input.toLowerCase() !== 'y') {
     process.exit(0);
   }
 
   const postOptions = {...options};
   for (const user of users) {
+    if (!use_passes) {
```

Comment:
You should check whether `user` is empty.

Sometimes a csv might have an empty row in the middle of the sheet.

When that happens the script still does the call for password change only to get the following error back from CouchDB.
```
ERROR  404 - {"error":"not_found","reason":"Database does not exist."}
```

It doesn't cause a crash but printing to screen a message like `Error: CSV row is empty` would be more intuitive.

Reply:
gotcha! fixed in 2c6cba7a0 . 

We now return a more clear `404`:

```
ERROR  404 - Username is blank - check CSV and run again.
```